[EA][PrivMon] Tile Visualisations on Privileged User Monitoring Dashboard

abhishekbhatia1710 opened the pull request:
## Summary

The PR adds code for displaying the visualisations for key insights panel of Privileged user monitoring dashboard.

It comprises of 6 tiles. 

1. Active Privileged Users
2. Alerts Triggered
3. Anomalies Detected
4. Granted Rights
5. Account Switches
6. Authentications

All the tiles have been created using the Lens visualisation for ease of use and also to streamline visualisations across the security solution.

Screenshots : 

Privileged User Monitoring Dashboard

![Screenshot 2025-06-09 at 3 00 55 PM](https://github.com/user-attachments/assets/f39768a2-bcd5-4959-ad53-b6186512ba49)


Lens visualisation for a tile : 

![image](https://github.com/user-attachments/assets/5e877124-ad6b-4cac-b9ef-12fa6a01b79e)


### Adding Data for desk testing : 

1. On the `main` branch of "The Data Yeeter" (https://github.com/elastic/security-documents-generator/), run `yarn start privileged_access_detection`. This primarily adds data for anomalies.
2. Then on the same `main` branch, run `yarn start privileged_user_monitoring`. This will add  data for the privileged user index
3. On the dev console execute the following : 

```
POST kbn:/api/entity_analytics/monitoring/engine/init

POST kbn:/api/entity_analytics/monitoring/users
{
  "user": {"name": "john.smith"}
}
POST kbn:/api/entity_analytics/monitoring/users
{
  "user": {"name": "stacy_armstrong"}
}
POST kbn:/api/entity_analytics/monitoring/users
{
  "user": {"name": "john_smith"}
}
POST kbn:/api/entity_analytics/monitoring/users
{
  "user": {"name": "randy.carlisle"}
}
POST kbn:/api/entity_analytics/monitoring/users
{
  "user": {"name": "root"}
}
```

### Testing Steps : 
1. Enable privilegedUserMonitoring feature flag.
2. Navigate to entity_analytics/privileged_user_monitoring page
3. Click on "Go to Dashboards" on the top left corner.
4. You will be able to see the tiles with name and number.
5. Click on the three dots when cursor is hovered over the tile and click on Inspect to check the query executed, click on More -> Open in Lens to check if the tile opens up fine in the lens visualiastion link. 
6. Check the data view in the lens visualisation. For anomalies, the data view should be `.ml-anomalies-*`. For other it would either be `.alerts-*` or `logs-*`.


### Not part of this PR : 
1. The trendline on the tile did not work as i am yet to figure out a way, if it exists, to show a trendline with ES|QL query as it works fine with KQL queries but similar lens attribites do not function.
2. Load testing where the local environment does not have much data to show. 


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

abhishekbhatia1710 committed:
Initial commit for the privmon key_insights_panel

abhishekbhatia1710 committed:
Merge branch 'main' into ea-12765-privmon-egg-box

abhishekbhatia1710 committed:
Merge branch 'main' into ea-12765-privmon-egg-box

abhishekbhatia1710 committed:
Changes for using ES|QL queries for tiles visualisation.

abhishekbhatia1710 committed:
Merge branch 'main' into ea-12765-privmon-egg-box

abhishekbhatia1710 committed:
Changes for tiles to display data

abhishekbhatia1710 committed:
Removing the "Open in Lens" option from the tiles

abhishekbhatia1710 committed:
Changes for lensAttribiutes and "open in lens" issues

abhishekbhatia1710 committed:
Interim changes for conditional data source availability

abhishekbhatia1710 committed:
Merge branch 'main' into ea-12765-privmon-egg-box

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/310031)
* Commit: 1160f8523204255bfd88650308ea1d71bd056d35

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/5c441a49cbd12a72a9f32373352dd524c154516d) | [after](https://github.com/elastic/kibana/commit/1160f8523204255bfd88650308ea1d71bd056d35) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 7637 | 7658 | +21 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/5c441a49cbd12a72a9f32373352dd524c154516d) | [after](https://github.com/elastic/kibana/commit/1160f8523204255bfd88650308ea1d71bd056d35) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 9.4MB | 9.4MB | +8.3KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/5c441a49cbd12a72a9f32373352dd524c154516d) | [after](https://github.com/elastic/kibana/commit/1160f8523204255bfd88650308ea1d71bd056d35) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 94.9KB | 94.9KB | +2.0B |



cc @abhishekbhatia1710

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310031","number":310031,"commit":"1160f8523204255bfd88650308ea1d71bd056d35"}],"number":310031}
buildkite-pr-comment-->

abhishekbhatia1710 committed:
Merge branch 'main' into ea-12765-privmon-egg-box

abhishekbhatia1710 committed:
Adding changes for account switch query to use TO_LOWER()

abhishekbhatia1710 committed:
Replacing process.command_line.caseless with to_lower(process.command_line)

abhishekbhatia1710 committed:
Merge branch 'main' into ea-12765-privmon-egg-box

abhishekbhatia1710 committed:
Unrequired code cleanup

abhishekbhatia1710 committed:
Merge branch 'ea-12765-privmon-egg-box' of https://github.com/abhishekbhatia1710/kibana into ea-12765-privmon-egg-box

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/311243)
* Commit: e9fdcf00708b28dc9aca1b335d6e1ccf58f5b0e7

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/81e362ed280fdfdee87d0dce4e76e08726ac8f16) | [after](https://github.com/elastic/kibana/commit/e9fdcf00708b28dc9aca1b335d6e1ccf58f5b0e7) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 7670 | 7691 | +21 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/81e362ed280fdfdee87d0dce4e76e08726ac8f16) | [after](https://github.com/elastic/kibana/commit/e9fdcf00708b28dc9aca1b335d6e1ccf58f5b0e7) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 9.4MB | 9.4MB | +8.3KB |



### History
* :green_heart: [Build #310329](https://buildkite.com/elastic/kibana-pull-request/builds/310329) succeeded 035a78ad2d17c18e9ab447c70d3e6121d4705e5e
* :green_heart: [Build #310031](https://buildkite.com/elastic/kibana-pull-request/builds/310031) succeeded 1160f8523204255bfd88650308ea1d71bd056d35

cc @abhishekbhatia1710

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311243","number":311243,"commit":"e9fdcf00708b28dc9aca1b335d6e1ccf58f5b0e7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310329","number":310329,"commit":"035a78ad2d17c18e9ab447c70d3e6121d4705e5e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310031","number":310031,"commit":"1160f8523204255bfd88650308ea1d71bd056d35"}],"number":311243}
buildkite-pr-comment-->

jaredburgettelastic committed:
Removed all of the usages of the custom timefilter code in the ES|QL queries, in favor of reusing the provided filters and passing that into the Lens component

jaredburgettelastic commented on x-pack/solutions/security/plugins/security_solution/public/entity_analytics/components/privileged_user_monitoring/components/key_insights_panel/alerts_triggered_tile/alerts_triggered_tile.tsx:
```diff
@@ -0,0 +1,62 @@
+  const lensAttributes = createKeyInsightsPanelLensAttributes({
+    title: 'Alerts Triggered',
+    label: 'Alerts Triggered',
+    esqlQuery: getAlertsTriggeredEsqlCount('default'),
```
space needed here

jaredburgettelastic commented on x-pack/solutions/security/plugins/security_solution/public/entity_analytics/components/privileged_user_monitoring/components/key_insights_panel/anomalies_detected_tile/anomalies_detected_tile.tsx:
```diff
@@ -0,0 +1,61 @@
+  const filterQuery = useEsqlGlobalFilterQuery();
+  const timerange = useGlobalTime();
+
+  const esqlQuery = getAnomaliesDetectedEsqlQuery('default');
```
space needed here

jaredburgettelastic commented on x-pack/solutions/security/plugins/security_solution/public/entity_analytics/components/privileged_user_monitoring/components/key_insights_panel/granted_rights_tile/granted_rights_tile.tsx:
```diff
@@ -0,0 +1,62 @@
+  const grantedRightsLensAttributes = createKeyInsightsPanelLensAttributes({
+    title: 'Granted Rights',
+    label: 'Granted Rights',
+    esqlQuery: getGrantedRightsEsqlCount('default'),
```
space needed here

jaredburgettelastic commented on x-pack/solutions/security/plugins/security_solution/public/entity_analytics/components/privileged_user_monitoring/components/key_insights_panel/account_switches_tile/account_switches_tile.tsx:
```diff
@@ -0,0 +1,70 @@
+      createKeyInsightsPanelLensAttributes({
+        title: 'Account Switches',
+        label: 'Account Switches',
+        esqlQuery: getAccountSwitchesEsqlCount('default'),
```
space needed here

jaredburgettelastic commented on x-pack/solutions/security/plugins/security_solution/public/entity_analytics/components/privileged_user_monitoring/components/key_insights_panel/active_privileged_users_tile/active_privileged_users_tile.tsx:
```diff
@@ -0,0 +1,62 @@
+  const lensAttributes = createKeyInsightsPanelLensAttributes({
+    title: 'Active Privileged Users',
+    label: 'Active Privileged Users',
+    esqlQuery: getActivePrivilegedUsersEsqlCount('default'),
```
space needed here

jaredburgettelastic commented on x-pack/solutions/security/plugins/security_solution/public/entity_analytics/components/privileged_user_monitoring/components/key_insights_panel/active_privileged_users_tile/esql_query.tsx:
```diff
@@ -0,0 +1,14 @@
+import { getPrivilegedMonitorUsersJoin } from '../../../helpers';
+
+export const getActivePrivilegedUsersEsqlCount = (namespace: string) => {
+  return `FROM logs-* METADATA _id, _index
+      ${getPrivilegedMonitorUsersJoin(namespace)}
+      | STATS COUNT(*)`;
```
This query needs to be adjusted. Right now, what it's doing is returning a count of all of the events within the time range. However, what we really need is the count of unique `user.name`s within that range.
Luckily, there's an extremely simple way to do this in ES|QL, suggestion below:

```suggestion
  return `FROM logs-* METADATA _id, _index
      ${getPrivilegedMonitorUsersJoin(namespace)}
      | STATS COUNT_DISTINCT(user.name)`;
```

jaredburgettelastic commented:
Overall, great job on this so far!!! It's so nice to see these visualizations come together.

Also desk tested, things looked great, with one exception: on a brand new installation, the anomalies visualization fails entirely, because there isn't an `.ml-anomalies-shared` index. It'd be nice to have it show `N/A` instead, although I won't block the PR for that reason. 

The other changes requested above **are** blockers by my estimation (the space awareness, and the "Active privileged users" query change). Please feel free to reach back out once those changes are made, and I'll approve!

abhishekbhatia1710 committed:
Get the space name dynamically to create ES|QL query as per the space name

abhishekbhatia1710 committed:
Consistent changes for space name

abhishekbhatia1710 committed:
Changed COUNT(*) to COUNT_DISTINCT(user.name)

abhishekbhatia1710 committed:
A different approach to use COUNT(*)

abhishekbhatia1710 committed:
Removing dataViewId from lensAttributes as it was not used

abhishekbhatia1710 committed:
Merge branch 'main' into ea-12765-privmon-egg-box

abhishekbhatia1710 committed:
Merge branch 'main' into ea-12765-privmon-egg-box

machadoum commented:
Hey @abhishekbhatia1710!

I left some comments, and I am primarily concerned about query duplication, especially since I am changing the queries in another PR. If you copy them, you will lose all the changes.

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/311842)
* Commit: 5e12934baa60c8393473f330cc8ea5d63fcc7e89

### Failed CI Steps
* [FTR Configs #27](https://buildkite.com/elastic/kibana-pull-request/builds/311842#0197a0b8-5b64-4875-933e-980aa17c613f)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/311842#0197a0b8-5b64-4875-933e-980aa17c613f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/311842/jobs/0197a0b8-5b64-4875-933e-980aa17c613f/artifacts/0197a0dc-9d0d-4585-b616-30e8bd0f2619) FTR Configs #<span></span>27 / Rules Management - Rule Bulk Action API @<span></span>ess @<span></span>serverless @<span></span>skipInServerless perform_bulk_action @<span></span>skipInServerlessMKI fill gaps run action should trigger the backfilling of the gaps for the rules in the request

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/8771e7f18b7b86e783096a54f4295568d4a449be) | [after](https://github.com/elastic/kibana/commit/5e12934baa60c8393473f330cc8ea5d63fcc7e89) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 7682 | 7702 | +20 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/8771e7f18b7b86e783096a54f4295568d4a449be) | [after](https://github.com/elastic/kibana/commit/5e12934baa60c8393473f330cc8ea5d63fcc7e89) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 9.4MB | 9.5MB | +7.7KB |



### History
* :yellow_heart: [Build #311467](https://buildkite.com/elastic/kibana-pull-request/builds/311467) was flaky 4d299c50a02040a00c377dd818a6089d06c161ff
* :yellow_heart: [Build #311320](https://buildkite.com/elastic/kibana-pull-request/builds/311320) was flaky bcb801bc78802bceb0561a2e89fa1d2d380d21f9
* :green_heart: [Build #311243](https://buildkite.com/elastic/kibana-pull-request/builds/311243) succeeded e9fdcf00708b28dc9aca1b335d6e1ccf58f5b0e7
* :green_heart: [Build #310329](https://buildkite.com/elastic/kibana-pull-request/builds/310329) succeeded 035a78ad2d17c18e9ab447c70d3e6121d4705e5e
* :green_heart: [Build #310031](https://buildkite.com/elastic/kibana-pull-request/builds/310031) succeeded 1160f8523204255bfd88650308ea1d71bd056d35

cc @abhishekbhatia1710

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311842","number":311842,"commit":"5e12934baa60c8393473f330cc8ea5d63fcc7e89"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311467","number":311467,"commit":"4d299c50a02040a00c377dd818a6089d06c161ff"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311320","number":311320,"commit":"bcb801bc78802bceb0561a2e89fa1d2d380d21f9"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311243","number":311243,"commit":"e9fdcf00708b28dc9aca1b335d6e1ccf58f5b0e7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310329","number":310329,"commit":"035a78ad2d17c18e9ab447c70d3e6121d4705e5e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310031","number":310031,"commit":"1160f8523204255bfd88650308ea1d71bd056d35"}],"number":311842}
buildkite-pr-comment-->

abhishekbhatia1710 commented on x-pack/solutions/security/plugins/security_solution/public/entity_analytics/components/privileged_user_monitoring/components/key_insights_panel/account_switches_tile/account_switches_tile.tsx:
```diff
@@ -0,0 +1,62 @@
+const LENS_VISUALIZATION_HEIGHT = 126;
+const LENS_VISUALIZATION_MIN_WIDTH = 160;
+
+export const AccountSwitchesTile: React.FC<{ spaceId: string }> = ({ spaceId }) => {
```
Sure, changed it to follow D.R.Y.

abhishekbhatia1710 committed:
Merge upstream/main and resolve conflicts

- Resolved modify/delete conflict for esql_source_query.ts
- Updated tile ESQL queries to use new structure from queries/helpers.ts
- Fixed imports in all tile components to use new helpers location
- Maintained refactored tile structure with KeyInsightsTile and i18n support

abhishekbhatia1710 committed:
Addressing review comments realted to lensattributes

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/312052)
* Commit: 00579450061c4657f4e0384018f0211ef24d7e3e

### Failed CI Steps
* [Build Kibana Distribution](https://buildkite.com/elastic/kibana-pull-request/builds/312052#0197a1dc-0eb0-4546-8d88-37a8e41a6c11)

### History
* :yellow_heart: [Build #311842](https://buildkite.com/elastic/kibana-pull-request/builds/311842) was flaky 5e12934baa60c8393473f330cc8ea5d63fcc7e89
* :yellow_heart: [Build #311467](https://buildkite.com/elastic/kibana-pull-request/builds/311467) was flaky 4d299c50a02040a00c377dd818a6089d06c161ff
* :yellow_heart: [Build #311320](https://buildkite.com/elastic/kibana-pull-request/builds/311320) was flaky bcb801bc78802bceb0561a2e89fa1d2d380d21f9
* :green_heart: [Build #311243](https://buildkite.com/elastic/kibana-pull-request/builds/311243) succeeded e9fdcf00708b28dc9aca1b335d6e1ccf58f5b0e7
* :green_heart: [Build #310329](https://buildkite.com/elastic/kibana-pull-request/builds/310329) succeeded 035a78ad2d17c18e9ab447c70d3e6121d4705e5e
* :green_heart: [Build #310031](https://buildkite.com/elastic/kibana-pull-request/builds/310031) succeeded 1160f8523204255bfd88650308ea1d71bd056d35

cc @abhishekbhatia1710

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312052","number":312052,"commit":"00579450061c4657f4e0384018f0211ef24d7e3e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311842","number":311842,"commit":"5e12934baa60c8393473f330cc8ea5d63fcc7e89"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311467","number":311467,"commit":"4d299c50a02040a00c377dd818a6089d06c161ff"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311320","number":311320,"commit":"bcb801bc78802bceb0561a2e89fa1d2d380d21f9"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311243","number":311243,"commit":"e9fdcf00708b28dc9aca1b335d6e1ccf58f5b0e7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310329","number":310329,"commit":"035a78ad2d17c18e9ab447c70d3e6121d4705e5e"}],"number":312052}
buildkite-pr-comment-->

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/312074)
* Commit: f480da80f06eeb880c18d0c5d6ff7fd404a8e1be

### Failed CI Steps
* [Build Kibana Distribution](https://buildkite.com/elastic/kibana-pull-request/builds/312074#0197a1f3-f2ea-49fd-81d8-fa7cd8765a46)

### History
* :broken_heart: [Build #312052](https://buildkite.com/elastic/kibana-pull-request/builds/312052) failed 00579450061c4657f4e0384018f0211ef24d7e3e
* :yellow_heart: [Build #311842](https://buildkite.com/elastic/kibana-pull-request/builds/311842) was flaky 5e12934baa60c8393473f330cc8ea5d63fcc7e89
* :yellow_heart: [Build #311467](https://buildkite.com/elastic/kibana-pull-request/builds/311467) was flaky 4d299c50a02040a00c377dd818a6089d06c161ff
* :yellow_heart: [Build #311320](https://buildkite.com/elastic/kibana-pull-request/builds/311320) was flaky bcb801bc78802bceb0561a2e89fa1d2d380d21f9
* :green_heart: [Build #311243](https://buildkite.com/elastic/kibana-pull-request/builds/311243) succeeded e9fdcf00708b28dc9aca1b335d6e1ccf58f5b0e7
* :green_heart: [Build #310329](https://buildkite.com/elastic/kibana-pull-request/builds/310329) succeeded 035a78ad2d17c18e9ab447c70d3e6121d4705e5e

cc @abhishekbhatia1710

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312074","number":312074,"commit":"f480da80f06eeb880c18d0c5d6ff7fd404a8e1be"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312052","number":312052,"commit":"00579450061c4657f4e0384018f0211ef24d7e3e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311842","number":311842,"commit":"5e12934baa60c8393473f330cc8ea5d63fcc7e89"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311467","number":311467,"commit":"4d299c50a02040a00c377dd818a6089d06c161ff"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311320","number":311320,"commit":"bcb801bc78802bceb0561a2e89fa1d2d380d21f9"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311243","number":311243,"commit":"e9fdcf00708b28dc9aca1b335d6e1ccf58f5b0e7"}],"number":312074}
buildkite-pr-comment-->

abhishekbhatia1710 committed:
Addressing review comment for shared queries and sourcerer data view

abhishekbhatia1710 committed:
Merge branch 'ea-12765-privmon-egg-box' of https://github.com/abhishekbhatia1710/kibana into ea-12765-privmon-egg-box

abhishekbhatia1710 commented:
Hi @machadoum, thank you for your feedback and suggestions!

I've updated the PR to incorporate the shared queries from your recently merged PR ([#223212](https://github.com/elastic/kibana/pull/223212/files)), and have also updated all ES|QL queries to use the Sourcerer data view in the FROM clause for consistency.

Additionally:

- Replaced repeated instances of KeyInsightsPanel with a single, reusable component
- Added i18n support for the tile titles and labels to ensure localisation

jaredburgettelastic committed:
Fixed alerts visualization to show alerts index instead of logs

jaredburgettelastic commented on x-pack/solutions/security/plugins/security_solution/public/entity_analytics/components/privileged_user_monitoring/components/key_insights_panel/account_switches_tile/esql_query.ts:
```diff
@@ -0,0 +1,16 @@
+import { getAccountSwitchesEsqlSource } from '../../../queries/account_switches_esql_query';
+
+export const getAccountSwitchesEsqlCount = (namespace: string, sourcerDataView: DataViewSpec) => {
+  const indexPattern = sourcerDataView?.title ?? '';
```
It'd be nice to have a better way to deal with potential null/undefined values here, however that can be done in a followup PR. In this case, what happens is the visualization just fails, but it would be better to just not render in those situations (so the `sourcerDataView` can always be truthy)

jaredburgettelastic reviewed (approved):
Desk tested and reviewed, no blockers from my end 👍

abhishekbhatia1710 commented on x-pack/solutions/security/plugins/security_solution/public/entity_analytics/components/privileged_user_monitoring/components/key_insights_panel/index.tsx:
```diff
@@ -0,0 +1,75 @@
+  return (
+    <EuiFlexGroup
+      wrap
+      css={css`
```
Thanks Pablo, new PR created for this change : https://github.com/elastic/kibana/pull/225307

elasticmachine commented:
Pinging @elastic/security-solution (Team: SecuritySolution)
